lib/ts/BufferWriterFormat.cc

```diff
@@ -403,6 +404,107 @@ namespace bw_fmt
     return w;
   }
 
+  /// Format for floating point values. Seperates floating point into a whole number and a
+  /// fraction. The fraction is converted into an unsigned integer based on the specified
+  /// precision, spec._prec. ie. 3.1415 with precision two is seperated into two unsigned
+  /// integers 3 and 14. The different pieces are assembled and placed into the BufferWriter.
+  /// The default is two decimal places. ie. X.XX. The value is always written in base 10.
+  ///
+  /// format: whole.fraction
+  ///     or: left.right
+  BufferWriter &
+  Format_Floating(BufferWriter &w, BWFSpec const &spec, double f, bool neg_p)
+  {
+    if (f == static_cast<uint64_t>(f)) { // integral
+      return Format_Integer(w, spec, static_cast<uint64_t>(f), neg_p);
+    }
+
+    double left, right;
+    size_t l = 0;
+    size_t r = 0;
+    char whole[std::numeric_limits<double>::digits + 1];
+    char fraction[std::numeric_limits<double>::digits + 1];
+    char dec      = '.';
+    char neg      = 0;
+    int width     = static_cast<int>(spec._min);                             // amount left to fill.
+    int precision = (spec._prec == BWFSpec::DEFAULT._prec) ? 2 : spec._prec; // default precision 2
+
+    right = modf(f, &left); // split the number
+
+    if (neg_p) {
+      neg = '-';
+    } else if (spec._sign != '-') {
+      neg = spec._sign;
+    }
+
+    // Convert trailing fraction into an integer value.
+    // Final precision point is always rounded.
+    uint64_t shift = 1;
```

Comment:
This seems wrong, it shouldn't be necessary. If later the digits will be extracted one at a time why not leave the multiplication to that spot?

Reply:
I tried to do this digit as their extracted. The issue I ran into was if a `float` was represented like 0.49999999 instead of 0.5. By multiplying, rounding and then casting it was ok. If we were to go digit by digit, we would still need to calculate some shift (by multiplication) and then round the fraction.

Reply:
You might look at precomputing the multipliers. There must be some maximum precision for a double and you could have a table of powers of 10 indexed by precision. There is a lot of pressure on performance in this code. Even if the power table were limited to say 10 powers, you could do this to get ones beyond the table and handle 99.9% of actual cases faster.